### PR TITLE
Speak-32 Add search algorithm

### LINKED ISSUE #32: Implement Search feature
We have a `Search` component, but we need to define the specs of how the search should work and implement it.

For an MVP version could be implemented on the front-end, since we have the whole list of speakers on the front-end and we can easily search the list without any queries to the database.

Later on, when we have pagination, we'll probably need to query the database to find the result(s).

### PR DESCRIPTION
**Pull request by tuul-wq**
Resolves #32 

**Description of changes in this PR**
Enabled `Search` component which has 2 autocomplete fields with a prepopulated dropdown.
Fields can be cleared with a cross icon. In case no speakers were found we show `NoResults `component with corresponding text (translation required). Also adjusted spinner center align as well as margin values to be similar with `NoResults `component to eliminate footer _jumping_. Moved `PaginationRow `in title row, to remove _jumping_ between search form and pagination widget. 

**How do the reviewers QA this?**
- [x] User can search by _topics_ or _locations_ or both of them.
- [x] Each search requests a new speakers list.
- [x] If filtered results contain more than 50 items, pagination will take place.
- [ ] Search applies on blur and clear events (from code perspective it's a `@change` event).

### CODE REVIEW

File: web/src/components/no-results/NoResults.vue [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<template>
+  <div class="pa-9 text-center">
+    <span class="display-1">{{ $t('findSpeaker.no-results') }}</span>
```
Comment (ann-kilzer) [inline web/src/components/no-results/NoResults.vue:3]: Let's avoid kebab-casing and stick with camelCase..

**Review by ann-kilzer — COMMENTED** [review]
Hey, thanks for adding this change!

We'd like to stick with camelCase and not introduce kebab-case, so please rename the no-results file as well as update the translation key.

Can you please write a little more details on your testing steps. Manual is fine, but detailing a few test cases you used helps us QA your changes more quickly. 🤠

**Comment by tuul-wq** [comment]
@ann-kilzer resolved kebab conflict; added more info for QA.

File: web/src/views/FindSpeaker.vue [hunk 2]
```diff
@@ -86,21 +90,24 @@ export default {
     Search,
     SpeakerCard,
     PaginationRow,
+    NoResults,
+  },
+  data() {
```
Comment (ann-kilzer) [inline web/src/views/FindSpeaker.vue:95]: (minor comment) There seem to be different opinions on which syntax is preferable. Actually this way looks pretty nice, though some of the VS Code plugins I use feel differently. 😄

File: web/src/i18n/locales/ja.json [hunk 3]
```diff
@@ -48,7 +48,9 @@
         "whyQ": "なぜ重要なのでしょうか？"
     },
     "findSpeaker": {
-        "title": "スピーカーを探す"
+        "title": "スピーカーを探す",
+        "pronouns": "Pronouns: {0}",
```
Comment (ann-kilzer) [inline web/src/i18n/locales/ja.json:52]: Please leave translation keys blank for ja.json for now, as we have the fallback locale of English. Thanks!

**Review by ann-kilzer — COMMENTED** [review]
This is looking great and it's going to be so helpful to be able to search. Pardon my delayed review. 

I'd like to request a few changes based on things I noticed while testing:

1) Since we only search by prefecture, let's change the Label "City, Prefecture, or Region" to "Prefecture."

2). I notice that when there is only one page of results, it appears possible to page to see more results. For example, search for "Blockchain" and see the following:

SpeakHer

It might be cool to allow searching on multiple Topics, using "OR" logic, but that's fine to happen in a later PR. Thanks again!

**Comment by tuul-wq** [comment]
@ann-kilzer updated PR

File: web/src/components/Search.vue [hunk 4]
```diff
@@ -74,17 +72,15 @@ export default {
       this.$db('Location')
         .select({
           view: 'All',
-          fields: ['prefecture', 'region'],
+          fields: ['prefecture'],
```
Comment (ann-kilzer) [inline web/src/components/Search.vue:75]: 🤔 I just noticed that the region column is in the DB, but there are no values. Probably best to add it later and fill out the DB.

File: web/src/views/FindSpeaker.vue [hunk 5]
```diff
@@ -92,31 +92,28 @@ export default {
...
+    isLoading: true,
+  }),
   computed: {
     selectedName() {
-      if (this.selectedSpeaker) {
-        if (this.$i18n.locale === 'ja') {
-          return this.selectedSpeaker.get('name_ja') || this.selectedSpeaker.get('name_en') || '';
-        }
-        return this.selectedSpeaker.get('name_en') || '';
+      if (!this.selectedSpeaker) return '';
```
Comment (ann-kilzer) [inline web/src/views/FindSpeaker.vue:111]: Can we keep this PR focused on search and avoid making other refactors? It gets confusing to follow the changes.

**Review by ann-kilzer — COMMENTED** [review]
Thanks for the updates. It looks like the pagination issue has been fixed. 👍🏻 

Almost ready to approve, but please address my comment and update with the latest master.

Please don't use force-push when contributing to this project, as it breaks the comment history within git, and can also cause issues if other people are working on the same branch. I recommend using merges. See our [contributing guidelines](https://github.com/WWCodeTokyo/speak-her-db/blob/master/CONTRIBUTING.md)

**Comment by tuttiq** [comment]
Hi @tuul-wq ! Any plans to come back to this PR?

**Review by ann-kilzer — APPROVED** [review]
Let's just get this in, it's going to make the site sooo much better. If we want to make follow ups, we can do it in future PRs. Thanks again @tuul-wq for the great work!

**Pull request merged**